app/controllers/albums_controller.rb

```diff
@@ -0,0 +1,67 @@
+class AlbumsController < ApplicationController
+  def index
+    @albums = Album.order(ranking: :desc)
+  end
+
+  def show
+    id = params[:id]
+    @album = Album.find(id)
+  end
+
+  def new
+    @album = Album.new
+  end
+
+  def upvote
+    @album = Album.find(params[:id])
+    if @album.ranking.nil?
```

Comment:
The whole method can be easily simplified. What if in ``create method`` we set default ranking value to 1, then you don't need to use ``if statement`` in here.